api/src/Gas.ts

```diff
@@ -91,39 +97,44 @@ export class GearGasSpent {
     sourceId: Hex,
     code: Hex | Buffer,
     payload: PayloadType,
-    value: number | string,
+    value?: number | string,
+    allowOtherPanics?: boolean,
     typeOfPayload?: string,
-  ): Promise<u64>;
+  ): Promise<GasInfo>;
 
   /**
    * Get gas spent of init message
    * @param sourceId Account id
    * @param code Program code
    * @param payload Payload of init message
    * @param value Value of message
+   * @param allowOtherPanics
    * @param metaOrTypeOfPayload Metadata or one of the primitives types
    * @returns number in U64 format
    */
   async init(
     sourceId: Hex,
     code: Hex | Buffer,
     payload: PayloadType,
-    value: number | string,
+    value?: number | string,
+    allowOtherPanics?: boolean,
     metaOrTypeOfPayload?: string | Metadata,
-  ): Promise<u64>;
+  ): Promise<GasInfo>;
 
   async init(
     sourceId: Hex,
     code: Hex | Buffer,
     payload: PayloadType,
-    value: number | string,
+    value?: number | string,
+    allowOtherPanics?: boolean,
     metaOrTypeOfPayload?: string | Metadata,
-  ): Promise<u64> {
-    return await this.api.rpc['gear'].getInitGasSpent(
+  ): Promise<GasInfo> {
+    return await this.api.rpc['gear'].calculateInitGas(
```

Comment:
```suggestion
    return this.api.rpc.gear.calculateInitGas(
```
`return await` is redudant

Reply:
I agree about await)